compat/src/index.d.ts

```diff
@@ -53,11 +53,11 @@ declare namespace React {
 
 	export function unmountComponentAtNode(container: Element | Document | ShadowRoot | DocumentFragment): boolean;
 
-	export function createFactory(type: preact.VNode["type"]): preact.VNode<{}>;
+	export function createFactory(type: preact.VNode["type"]): (props?: any, ...children: preact.ComponentChildren[]) => preact.VNode<any>;
 	export function isValidElement(element: any): boolean;
 	export function findDOMNode(component: preact.Component): Element | null;
 
-	export interface PureComponent<P = {}, S = {}> extends preact.Component {
+	export class PureComponent<P = {}, S = {}> extends preact.Component {
```

Comment:
`PureComponent` is an actual class that should be extended by other classes, not an interface that any object implement